Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -213,48 +242,34 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate {
 
 extension SearchViewController: UITableViewDataSource {
     override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
-        let cell = super.tableView(tableView, cellForRowAtIndexPath: indexPath)
-
-        if let currentSection = SearchListSection(rawValue: indexPath.section) {
-            switch currentSection {
-            case .SearchSuggestions:
-                cell.textLabel?.text = searchSuggestions[indexPath.row]
-                cell.detailTextLabel?.text = nil
-                cell.imageView?.image = nil
-                cell.isAccessibilityElement = false
-                cell.accessibilityLabel = nil
-            case .BookmarksAndHistory:
-                if let site = data[indexPath.row] as? Site {
-                    cell.textLabel?.text = site.title
-                    cell.detailTextLabel?.text = site.url
-                    if let img = site.icon? {
-                        let imgUrl = NSURL(string: img.url)
-                        cell.imageView?.sd_setImageWithURL(imgUrl, placeholderImage: self.profile.favicons.defaultIcon)
-                    } else {
-                        cell.imageView?.image = self.profile.favicons.defaultIcon
-                    }
-                    cell.isAccessibilityElement = false
-                    cell.accessibilityLabel = nil
+        switch SearchListSection(rawValue: indexPath.section)! {
```

Comment:
This makes no sense at all to me. A crasher in something as high level and important as the search view controller will result in an unusable application that the user will likely uninstall and never look back at. An incorrectly rendered UI on the other hand is still something that we can notice and deal with. And most retain the user.

Maybe crashing is a common practice in Android land, where it results in a stack trace that you can record and recover from.

_On iOS it results in a hard crash. The app exits, the user loses their session._

We really need to take advantage of swift's safety belts and not ignore them. If we are afraid that we miss cases where things go wrong then we need to find an alterntive reporting mechanism. Crashing is not it. Instead we could log or fhr or something better.

I'm happy with assertions in debug builds.